File: app/src/main/kotlin/com/example/aura/shared/model/VideoUi.kt
package com.example.aura.shared.model

import androidx.compose.runtime.Immutable
import com.example.aura.domain.model.Video
import kotlinx.serialization.Serializable

@Immutable
@Serializable
data class VideoUi(
    val id: Long,
    val width: Int,
    val height: Int,
    val duration: Int,
    val imageUrl: String,
    val videoUrl: String,
    val photographer: String,
    val isFavorite: Boolean
)


fun Video.toUi(): VideoUi {
    return VideoUi(
        id = id,
        width = width,
        height = height,
        duration = duration,
        imageUrl = image,
        videoUrl = videoUrl,
        photographer = user.name,
        isFavorite = isFavorite
    )
}

File: iosApp/iosApp/core/model/VideoUi.swift
import Foundation
import Shared

struct VideoUi: Identifiable, Equatable, Hashable {
    let id: Int64
    let videoUrl: String
    let thumbnailUrl: String
    let duration: Int
    let width: Int
    let height: Int
    let photographerName: String
}

extension Video {
    func toUi() -> VideoUi {
        return VideoUi(
            id: self.id,
            videoUrl: self.videoUrl,
            thumbnailUrl: self.imageUrl,
            duration: Int(self.duration),
            width: Int(self.width),
            height: Int(self.height),
            photographerName: self.user.name
        )
    }
}
